### PR TITLE
[WIP] delete results when copy or archive

### PR DESCRIPTION
**Pull request by berthaultval**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*
When you copy or archive, all results are kept.

**What is the new behavior (if this is a feature change)?**
When you copy, results are deleted and when you archive you have an option for keep or not the results.

### CODE REVIEW

File: afs-core/src/main/java/com/powsybl/afs/AbstractNodeBase.java [hunk 1]
```diff
@@ -219,8 +216,12 @@ public void unarchive(Path dir, boolean isZipped) {
         }
     }
 
+    public void archive(Path dir, Map<String, List<String>> outputBlackList) {
+        archive(dir, false, false, outputBlackList);
+    }
+
     public void archive(Path dir) {
-        archive(dir, false, false);
+        archive(dir, false, false, new HashMap<>());
```
Comment (pl-buiquang) [inline afs-core/src/main/java/com/powsybl/afs/AbstractNodeBase.java]: use Collections.emptyMap()

File: afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java [hunk 2]
```diff
@@ -99,6 +106,11 @@ public Options getOptions() {
                         .desc("archive dependencies")
                         .hasArg(false)
                         .build());
+                topLevelOptions.addOption(Option.builder("keepResults")
+                        .longOpt(RESULT)
+                        .desc("keep results of MS and metrix")
```
Comment (pl-buiquang) [inline afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java]: MS and metrix objetcs don't exist here. 
Also this should not be a top level option.
The option should be to clear the result and not keep them, in order to have the same behavior as before.

File: afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java [hunk 3]
```diff
@@ -99,6 +106,11 @@ public Options getOptions() {
                         .desc("archive dependencies")
                         .hasArg(false)
                         .build());
+                topLevelOptions.addOption(Option.builder("keepResults")
```
Comment (pl-buiquang) [inline afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java]: set this as a static string (also other options do not have this set..)

File: afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java [hunk 4]
```diff
@@ -33,6 +37,7 @@
     private static final String UNARCHIVE = "unarchive";
     private static final String ZIP = "zip";
     private static final String DEPENDENCIES = "dependencies";
+    private static final String RESULT = "results";
```
Comment (pl-buiquang) [inline afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java]: not very meaningful

File: afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java [hunk 5]
```diff
@@ -191,7 +203,13 @@ private void runArchive(CommandLine line, ToolRunningContext context) {
             Path dir = context.getFileSystem().getPath(line.getOptionValue(DIR));
             boolean mustZip = line.hasOption(ZIP);
             boolean archiveDependencies = line.hasOption(DEPENDENCIES);
-            fs.getRootFolder().archive(dir, mustZip, archiveDependencies);
+            boolean keepResults = line.hasOption(RESULT);
+            Map<String, List<String>> outputBlackList = new HashMap<>();
+            if (!keepResults) {
```
Comment (pl-buiquang) [inline afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java]: invert the condition

File: afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageArchive.java [hunk 6]
```diff
@@ -105,10 +105,22 @@ public String getIdMapping(String nodeId) {
 
         private final List<String> idListObject = new ArrayList<>();
 
+        private final Map<String, List<String>> outputBlackList = new HashMap();
+
         private final boolean archiveDependencies;
 
         public ArchiveContext(boolean archiveDependencies) {
             this.archiveDependencies = archiveDependencies;
+            this.outputBlackList.putAll(new HashMap<>());
```
Comment (pl-buiquang) [inline afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageArchive.java]: useless

File: afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageArchive.java [hunk 7]
```diff
@@ -152,7 +164,16 @@ private void writeDependencies(NodeInfo nodeInfo, Path nodeDir) throws IOExcepti
         }
     }
 
-    private void writeData(NodeInfo nodeInfo, Path nodeDir) throws IOException {
+    private boolean checkResultBlackList(Map<String, List<String>> blackMap, String dataName, String pseudoClass) {
+        List<String> blackList = blackMap.get(pseudoClass);
+        return blackList == null ? true : blackList.stream()
```
Comment (pl-buiquang) [inline afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageArchive.java]: simplify : blackList == null || blackList.stream().noneMatch(element -> element.contains(dataName))

File: afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java (hunk 4 above)
Comment (berthaultval) [reply afs-core/src/main/java/com/powsybl/afs/AppFileSystemTool.java]: sonar -> code smell

**Pull request closed without merging**